src/SignalR/common/Shared/AsyncEnumerableAdapters.cs

```diff
@@ -12,16 +12,46 @@ namespace Microsoft.AspNetCore.SignalR.Internal
     // True-internal because this is a weird and tricky class to use :)
     internal static class AsyncEnumerableAdapters
     {
+#if NETCOREAPP3_0
         public static IAsyncEnumerable<object> MakeCancelableAsyncEnumerable<T>(IAsyncEnumerable<T> asyncEnumerable, CancellationToken cancellationToken = default)
         {
             return new CancelableAsyncEnumerable<T>(asyncEnumerable, cancellationToken);
         }
 
+        public static IAsyncEnumerable<T> MakeCancelableTypedAsyncEnumerable<T>(IAsyncEnumerable<T> asyncEnumerable, CancellationTokenSource cts)
+        {
+            return new CancelableTypedAsyncEnumerable<T>(asyncEnumerable, cts);
+        }
+
         public static IAsyncEnumerable<object> MakeCancelableAsyncEnumerableFromChannel<T>(ChannelReader<T> channel, CancellationToken cancellationToken = default)
         {
             return MakeCancelableAsyncEnumerable(channel.ReadAllAsync(), cancellationToken);
         }
 
+        private class CancelableTypedAsyncEnumerable<TResult> : IAsyncEnumerable<TResult>
+        {
+            private readonly IAsyncEnumerable<TResult> _asyncEnumerable;
+            private readonly CancellationTokenSource _cts;
+
+            public CancelableTypedAsyncEnumerable(IAsyncEnumerable<TResult> asyncEnumerable, CancellationTokenSource cts)
+            {
+                _asyncEnumerable = asyncEnumerable;
+                _cts = cts;
+            }
+
+            public IAsyncEnumerator<TResult> GetAsyncEnumerator(CancellationToken cancellationToken = default)
+            {
+                if (cancellationToken.CanBeCanceled)
+                {
+                    cancellationToken.Register(() =>
+                    {
+                        _cts.Cancel();
+                    });
+                }
+                return _asyncEnumerable.GetAsyncEnumerator(cancellationToken);
```

Comment:
No need to pass in the cancellationToken here. If we know it won't be used, it's a little misleading to pass it in.